src/languages/nix.js

```diff
@@ -9,85 +9,291 @@ Category: system
+      "currentSystem",
+      "currentTime",
+      "deepSeq",
+      "derivation",
+      "derivationStrict",
+      "dirOf",
+      "div",
+      "elem",
+      "elemAt",
+      "false",
+      "fetchGit",
+      "fetchMercurial",
+      "fetchTarball",
+      "fetchTree",
+      "fetchurl",
+      "filter",
+      "filterSource",
+      "findFile",
+      "flakeRefToString",
+      "floor",
+      "foldl'",
+      "fromJSON",
+      "fromTOML",
+      "functionArgs",
+      "genList",
+      "genericClosure",
+      "getAttr",
+      "getContext",
+      "getEnv",
+      "getFlake",
+      "groupBy",
+      "hasAttr",
+      "hasContext",
+      "hashFile",
+      "hashString",
+      "head",
+      "import",
+      "intersectAttrs",
+      "isAttrs",
+      "isBool",
+      "isFloat",
+      "isFunction",
+      "isInt",
+      "isList",
+      "isNull",
+      "isPath",
+      "isString",
+      "langVersion",
+      "length",
+      "lessThan",
+      "listToAttrs",
       "map",
+      "mapAttrs",
+      "match",
+      "mul",
+      "nixPath",
+      "nixVersion",
+      "null",
+      "parseDrvName",
+      "parseFlakeRef",
+      "partition",
+      "path",
+      "pathExists",
+      "placeholder",
+      "readDir",
+      "readFile",
+      "readFileType",
       "removeAttrs",
+      "replaceStrings",
+      "scopedImport",
+      "seq",
+      "sort",
+      "split",
+      "splitVersion",
+      "storeDir",
+      "storePath",
+      "stringLength",
+      "sub",
+      "substring",
+      "tail",
       "throw",
+      "toFile",
+      "toJSON",
+      "toPath",
       "toString",
-      "derivation"
-    ]
+      "toXML",
+      "trace",
+      "traceVerbose",
+      "true",
+      "tryEval",
+      "typeOf",
+      "unsafeDiscardOutputDependency",
+      "unsafeDiscardStringContext",
+      "unsafeGetAttrPos",
+      "warn",
+      "zipAttrsWith",
+    ].map(b => `builtins\\.${b}`).join('|') + ')'),
   };
-  const ANTIQUOTE = {
-    className: 'subst',
-    begin: /\$\{/,
-    end: /\}/,
-    keywords: KEYWORDS
+
+  const IDENTIFIER_REGEX = '[A-Za-z_][A-Za-z0-9_\'-]*';
+
+  const LOOKUP_PATH = {
+    className: 'symbol',
+    match: new RegExp(`<${IDENTIFIER_REGEX}(/${IDENTIFIER_REGEX})*>`),
   };
-  const ESCAPED_DOLLAR = {
-    className: 'char.escape',
-    begin: /''\$/,
+
+  const PATH = {
+    className: 'symbol',
+    match: new RegExp(`(\\.\\.|\\.|~)?/(${IDENTIFIER_REGEX})?(/${IDENTIFIER_REGEX})*`),
+  };
+
+  const OPERATOR = {
+    className: 'operator',
+    relevance: 0,
+    match: /(==|=|\+\+|\+|<=|<\||<|>=|>|->|-|\/\/|\/|!=|!|\|\||\|>|\?|\*|&&)/,
   };
+
   const ATTRS = {
-    begin: /[a-zA-Z0-9-_]+(\s*=)/,
+    begin: new RegExp(`${IDENTIFIER_REGEX}(\\s*=)`),
     returnBegin: true,
     relevance: 0,
     contains: [
       {
         className: 'attr',
         begin: /\S+/,
         relevance: 0.2
-      }
-    ]
+      },
+    ],
+  };
+
+  const NORMAL_ESCAPED_DOLLAR = {
+    className: 'char.escape',
```

Comment:
Please use the new `scope` everywhere in place of `className` since this is such a big rewrite.